### ISSUE TITLE
Multiple port and site support

### ISSUE THREAD

**Issue by mc388** [containous/traefik#61]
It would be nice if traefik can handle multiple ports and connect different web applications to different port, such as [docker-gen](https://github.com/jwilder/docker-gen) works.

At the moment I work wich `docker-gen` and my docker-compose file from each of my application looks like this:

```
web:
  image: nginx:latest
  ports:
    - "50020:80"
    - "50021:443"
  environment:
    VIRTUAL_HOST: demo-project
  links:
    - "php"
  volumes:
    - ./logs/nginx:/var/log/nginx
    - ./nginx/ssl:/etc/nginx/ssl
  volumes_from:
    - php

php:
  image: mc388/docker-php:latest
  ...
```

The `docker-gen` maps this application to port 50020 and 50021. Via the `VIRTUAL_HOST` parameter the proxy knows to which port the request is forwarded. 

An other feature could be to config separate ssl certs for each web application.

If traefik can handle something like this, it would be very nice.

**Comment by F0rth** [comment]
+1 for the ssl certs per frontends.

**Comment by emilevauge** [comment]
I think you should be able to put multiple certs in the same file with different domains right now, and traefik should choose the good one. But I need to test it.

**Comment by benschumacher** [comment]
@emilevauge Absolutely TLS SNI is a must. Should be relatively straight-forward w/Go.

**Comment by blackbarn** [comment]
A way to define multiple host and port mappings for a single container would be very helpful.

**Comment by jas99** [comment]
@emilevauge Any plans on this brother?

**Comment by ylecuyer** [comment]
@emilevauge ping

**Comment by emilevauge** [comment]
There is a proposal here #444. You are welcome to discuss on the rule format evolution :)

**Comment by rafilkmp3** [comment]
Still nothing ? This one port limit is holds my migration from  nginx gen to traefik

**Comment by ldez** [comment]
Closed by #1257
